docs/source/providers/index.md

```diff
@@ -0,0 +1,45 @@
+# Providers Overview
```

Comment:
I like this. What do you think @raghotham @hardikjshah?

Reply:
Hey @ashwinb I updated things a little bit so that it's easier to navigate. I also added some initial copy to have coverage for all of the current providers but some of them could use a lot more love than others (I think it'll take time to actually do it right, but having a place where that can be contributed to feels useful).

I figured this would be a good start so that if other providers are added, we can request that they update the docs.  

LMK if you want some changes but I figured this was a good enough MVP so that I can start documenting some of the performance trade-offs between sqlite-vec and faiss as outlined here: https://github.com/meta-llama/llama-stack/issues/1165.

Reply:
I'm fine with adding this documentation - but I'm worried about leaving placeholders which will make the page look incomplete. I'd rather we add at minimum some blurb about each section.

Reply:
happy to! I'll put together what makes sense, feel free to comment on any sections that you have any opinions on and I'll incorporate.

Reply:
@raghotham I've added a short blurb that's mostly a copy paste from the Core Concepts section: https://llama-stack.readthedocs.io/en/latest/concepts/index.html#resources